lib/posix/timer.c

```diff
@@ -3,11 +3,31 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  */
+
+#ifdef CONFIG_PICOLIBC
+/* FIXME: Please see Issue #46910 */
+#define _POSIX_MONOTONIC_CLOCK
```

Comment:
You probably want to add `#define _POSIX_TIMERS 1` here -- that way you'll get things like `clock_gettime`